File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const [N, M] = input[0].split(' ').map(Number);
const steps = input.slice(1);
const baskets = Array.from(new Array(N), (_, i) => i + 1);

for (let step of steps) {
    const [i, j] = step.split(' ').map(x => +x - 1);
    
    baskets.splice(i, j - i + 1, ...baskets.slice(i, j + 1).reverse());
}

console.log(baskets.join(' '));

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.py
import sys

N, M = map(int, input().split())
baskets = [i + 1 for i in range(N)]

for line in sys.stdin:
    i, j = map(int, line.rstrip().split())
    part = baskets[i-1:j]
    part.reverse()
    
    for bi, pi in zip(range(i-1, j), range(len(part))):
        baskets[bi] = part[pi]

print(' '.join(map(str, baskets)))